File: programmers/lv0/코딩기초트레이닝/연산과조건문/홀짝에따라다른값반환하기/Solution.py
def solution(n):
    answer = 0
    isEven = n % 2 == 0
    start = 2 if isEven else 1
    for i in range(start, n + 1, 2):
        answer += i * i if isEven else i
    return answer

# simple
# def solution(n):
#     if n%2:
#         return sum(range(1,n+1,2))
#     return sum([i*i for i in range(2,n+1,2)])

print(solution(7))
print(solution(10))

File: programmers/lv0/코딩기초트레이닝/연산과조건문/홀짝에따라다른값반환하기/Solution.js

const solution = n => {
    let answer = 0
    const isEven = n % 2 === 0
    let start = isEven ? 2 : 1
    for (let i = start; i <= n; i += 2) {
        answer += isEven ? i * i : i
    }
    return answer
}